main.sh

```diff
@@ -61,15 +61,16 @@ cluster_up() {
 	local kubeconfig
 	kubeconfig="$("${CLUSTER_PROVIDER}"_kubeconfig)"
 
-	mkdir -p "$(basename "$KUBECONFIG_ROOT_DIR"/)"
-	mv -f "$kubeconfig" "$KUBECONFIG_ROOT_DIR"/"$KEPLER_KUBECONFIG"
-	
-	kubeconfig=$(find "$KUBECONFIG_ROOT_DIR" -type f -name "*config*" | tr '\n' ':') 
-	KUBECONFIG=$kubeconfig kubectl config view --flatten > all-in-one-kubeconfig.yaml
-	mv -f all-in-one-kubeconfig.yaml "$KUBECONFIG_ROOT_DIR"/config
-	export KUBECONFIG="$KUBECONFIG_ROOT_DIR"/config
+	mkdir -p "$(basename "$KUBECONFIG_ROOT_DIR")"
+	mv -f "$kubeconfig" "${KUBECONFIG_ROOT_DIR}/${KEPLER_KUBECONFIG}"
 
-	kubectl config --kubeconfig="$KUBECONFIG" set-cluster kind-kepler
+	kubeconfig="$KUBECONFIG_ROOT_DIR/config":$(find "$KUBECONFIG_ROOT_DIR" \
+		-type f -name "*config*" | tr '\n' ':')
+	kubeconfig=${kubeconfig%:}
+	KUBECONFIG=$kubeconfig kubectl config view --merge --flatten >all-in-one-kubeconfig.yaml
```

Comment:
if it's all in one and later, we shut down current cluster, will we remove the cluster access to tidy the all in one confg yaml file?

Reply:
Yes, when we bring the current cluster down it cleans up the merged config.
When cluster is up:
```
-rw-r--r--@ 1 vibhuprashar  staff  12242 Nov 22 16:56 kubeconfig
-rw-r--r--  1 vibhuprashar  staff   5583 Nov 23 22:04 config-kepler
drwxr-x---  4 vibhuprashar  staff    128 Nov 23 22:04 cache
-rw-r--r--  1 vibhuprashar  staff  17730 Nov 23 22:05 config
```
When cluster is down:
```
-rw-r--r--@ 1 vibhuprashar  staff  12242 Nov 22 16:56 kubeconfig
drwxr-x---  4 vibhuprashar  staff    128 Nov 23 22:04 cache
-rw-r--r--  1 vibhuprashar  staff  12219 Nov 23 22:06 config
```